### PR TITLE
Add default-branch sync guard for PR workflows

### PR DESCRIPTION
**Pull request by KjellKod**
## ⭐ Why this matters
PR branches can drift behind the default branch before they are opened or shepherded, which leaves reviewers discovering avoidable conflicts late. This adds one shared guard so the PR workflow checks freshness first, rebases only when clean, and stops on conflicts without overwriting default-branch work.

---

## Summary
Add a shared default-branch sync helper and wire it into the PR assistant and PR shepherd guidance before PR creation or shepherd pushes.
- The helper fetches `origin`, detects the remote default branch, reports up-to-date/clean/conflict states, and applies clean rebase or merge syncs without pushing.
- The completed Quest idea is archived into the journal and moved out of active ideas.

## Changes
- **PR workflow**:
  - Add `scripts/pr_sync_default_branch.py` to inspect or apply remote default-branch syncs with JSON output.
  - Update `.skills/pr-assistant/SKILL.md` to run the helper before push or `gh pr create`, including the narrow `--force-with-lease` rule for clean own-branch rebases.
  - Update `.skills/pr-shepherd/SKILL.md` to run the helper in merge mode before shepherd commits or pushes so open PR review anchors are preserved.
- **Conflict safety**:
  - Treat merge-tree fatal errors separately from real conflicts.
  - Abort failed rebase or merge attempts and return conflicted files instead of leaving the worktree mid-operation.
  - Document that blanket `-X theirs` and `-X ours` conflict resolution is not allowed.
- **Quest archive**:
  - Move the pre-PR freshness idea from active ideas to done.
  - Add the Quest journal entry, celebration artifact, and manifest entry.
- **Tests**:
  - Add focused unit coverage for default-branch detection, inspect/apply behavior, conflict reporting, apply-time aborts, merge-tree fatal errors, and parser edge cases.

## Validation
- [x] Run `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py`. Expected result: all 68 focused tests pass, covering the helper and the shepherd workflow changes.
- [x] Run `bash scripts/quest_validate-manifest.sh`. Expected result: manifest validation reports all Quest files listed and no stale entries.
- [x] Run `python3 scripts/pr_sync_default_branch.py --json` from this PR branch after fetching. Expected result on the current branch: JSON with `status: "up_to_date"`, `ok: true`, and `default_branch: "main"`.
Watch for: the inspect path uses `git merge-tree` as a best-effort estimate; `--apply` remains the source of truth for the actual rebase or merge operation and must stop on conflicts.

## Notes
- During PR preparation, the new helper correctly detected an additive conflict in `docs/quest-journal/README.md`; the branch was then manually resolved by preserving the current `origin/main` timeline and adding the new `pre-pr-sync` row above it.

<pre>
     ▐▛███▜▌
    ▝▜█████▛▘
      ▘▘ ▝▝
Quest/Co-Authored by
Co-Authored-By: Claude <noreply@anthropic.com>
Co-Authored-By: Codex <noreply@openai.com>
in collaboration with KjellKod
</pre>

### CODE REVIEW

File: scripts/pr_sync_default_branch.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+            {
+                "ok": True,
+                "status": STATUS_CLEAN,
+                "action": action,
+                "reason": "",
+            }
+        )
+        return 0, payload
+
+    apply_code, apply_message = _apply_sync(strategy, default_ref)
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_sync_default_branch.py:290]: Fixed in c84c090. Apply mode now checks for a dirty worktree and in-progress merge/rebase/cherry-pick state before calling rebase or merge; unsafe states return `status: "error"` without mutating the branch. I also updated the shepherd skill text so sync applies only from a clean worktree before push. Validation: `python3 -m pytest tests/unit/` passed with 694 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: scripts/pr_sync_default_branch.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    print(json.dumps(payload, sort_keys=True))
+    return code
+
+
+def _message(result: subprocess.CompletedProcess[str]) -> str:
+    return (result.stderr or result.stdout).strip()[:500]
+
+
+def detect_default_branch() -> tuple[str, str]:
+    symbolic = _run(["git", "symbolic-ref", "refs/remotes/origin/HEAD"])
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_sync_default_branch.py:79]: Fixed in c84c090. The helper now discovers the remote default branch via `git ls-remote --symref origin HEAD` before falling back to local `origin/HEAD` and then `gh`, so a stale local symbolic ref cannot silently select the wrong branch. Validation: `python3 -m pytest tests/unit/` passed with 694 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: .skills/pr-assistant/SKILL.md [hunk 3]
```diff
@@ -10,6 +10,39 @@ Generate a pull request title and description from the current branch, then crea
...
+
+Before pushing or running `gh pr create`, sync the PR branch with the remote
+default branch from the source worktree:
+
+1. Run `python3 scripts/pr_sync_default_branch.py --json`.
+2. If the payload is `status: "up_to_date"`, continue with the normal PR
+   creation flow.
+3. If the payload is `status: "clean"`, run
+   `python3 scripts/pr_sync_default_branch.py --apply --json`, then continue.
+   The inspect result is a best-effort merge-based estimate; the `--apply`
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .skills/pr-assistant/SKILL.md]: Fixed in 40c8148. The pr-assistant sync step now requires parsing the apply payload and continuing only for `synced` or `up_to_date`; `conflict` surfaces files/hunks, and `error` stops with reason/message before any push or PR mutation. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 71 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: .skills/pr-assistant/SKILL.md [hunk 4]
```diff
@@ -10,6 +10,39 @@ Generate a pull request title and description from the current branch, then crea
...
+   `conflict_files` and the specific hunks to the human. Attempt a resolution
+   only when it is clearly safe and non-destructive: additive / adjacent /
+   whitespace / import-or-list ordering, where both sides' intent is preserved
+   and nothing from `main` is dropped. If the correct resolution would overwrite
+   or delete code/instructions from `main`, or the right resolution is
+   ambiguous, pause and ask the human. When in doubt, pause.
+5. Never use blanket `-X theirs` or `-X ours` resolution.
+6. If the applied sync reports `force_with_lease: true`, push with
+   `git push --force-with-lease`; otherwise push normally.
+
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .skills/pr-assistant/SKILL.md:43]: Fixed in 40c8148. The force-with-lease guidance now uses `git push --force-with-lease -u origin HEAD` for a first push and reserves plain `git push --force-with-lease` for branches that already have an upstream. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 71 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: scripts/pr_sync_default_branch.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+    guard_reason, guard_message = _pre_apply_error()
+    if guard_reason:
+        payload.update({"status": STATUS_ERROR, "reason": guard_reason})
+        if guard_message:
+            payload["message"] = guard_message
+        return 1, payload
+
+    apply_code, apply_message = _apply_sync(strategy, default_ref)
+    if apply_code != 0:
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_sync_default_branch.py:291]: Fixed in ed010b9. Apply failures now return `status: "conflict"` only when unmerged paths exist; failures without conflicted files return `status: "error"` with `reason: "rebase_failed"` or `"merge_failed"` and the apply message. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 72 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: scripts/pr_sync_default_branch.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+        if result.returncode == 0:
+            return reason, ""
+
+    return "", ""
+
+
+def sync(strategy: str = "rebase", *, apply: bool = False) -> tuple[int, dict[str, Any]]:
+    payload = _base_payload(strategy)
+
+    fetch = _run(["git", "fetch", "origin"])
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_sync_default_branch.py:217]: Fixed in e5c5af8. Before apply-mode rebase can return a force-with-lease path, the helper now checks whether the local branch contains its upstream after fetch. If remote-only branch commits exist, it returns `status: "error"` with `reason: "upstream_not_contained"` instead of allowing a later lease push to clobber them. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 73 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: scripts/pr_sync_default_branch.py [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+        result = _run(["git", "rev-parse", "--verify", "-q", ref])
+        if result.returncode == 0:
+            return reason, ""
+
+    return "", ""
+
+
+def _pre_rebase_lease_error() -> tuple[str, str]:
+    upstream = _run(["git", "rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{u}"])
+    if upstream.returncode != 0 or not upstream.stdout.strip():
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_sync_default_branch.py]: Fixed in 5e72b12. When no upstream is configured, the helper now checks for an existing same-named `origin/<branch>` ref after fetch and refuses rebase/lease flow if local HEAD does not contain it. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 74 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: scripts/pr_sync_default_branch.py [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+
+def probe_merge(default_ref: str) -> tuple[str, list[str], str]:
+    result = _run(
+        [
+            "git",
+            "merge-tree",
+            "--write-tree",
+            "--no-messages",
+            "-z",
+            "--name-only",
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/pr_sync_default_branch.py:134]: Fixed in 5e72b12. The workflow docs now explicitly say inspect conflicts may be filename-only; if hunks or conflict markers are unavailable, the agent must pause instead of guessing. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 74 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod

File: .skills/pr-shepherd/SKILL.md [hunk 9]
```diff
@@ -57,6 +57,28 @@ workspace still matches the PR branch:
    is that worktree before committing.
 6. If branch or workspace verification fails, stop and ask the user to confirm
    the intended workspace before editing, committing, or pushing.
+7. After workspace verification, sync with the remote default branch from this
+   PR worktree using the shared helper only when the worktree is clean. If the
+   worktree is dirty because a local fix is being prepared, do not apply sync
+   over those edits; finish and validate the fix commit first, then rerun this
+   guard before pushing:
+   - For normal shepherding of an already-open PR, run
+     `python3 scripts/pr_sync_default_branch.py --strategy merge --apply --json`.
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .skills/pr-shepherd/SKILL.md:66]: Fixed in f6733b5. The shepherd workflow now has an explicit `status: "error"` branch for the sync helper: stop before any commit or push, surface `reason`/`message`, and do not continue from a stale or partially checked sync state. Validation: `python3 -m pytest tests/unit/test_pr_sync_default_branch.py tests/unit/test_pr_shepherd.py` passed with 74 tests.\n\n- Reviewed by Codex, in collaboration with KjellKod
